app/controllers/katello/api/v2/packages_controller.rb

```diff
@@ -62,6 +62,38 @@ def custom_index_relation(collection)
       collection
     end
 
+    api :GET, "/content_view_versions/:id/available_packages", N_("List packages that can be added to the Content View Version via an Incremental Update")
+    param :id, :number, :desc => N_("Content View Version identifier"), :required => true
+    param :repository_id, :number, :desc => N_("repository identifier")
+    param :organization_id, :number, :desc => N_("organization identifier to list applicable packages for")
+    param :host_id, :number, :desc => N_("host id to list applicable packages for")
+    param :packages_restrict_applicable, :bool, :desc => N_("show only packages with one or more applicable hosts")
+    param :packages_restrict_upgradable, :bool, :desc => N_("show only packages with one or more upgradable hosts")
+    param_group :search, ::Katello::Api::V2::ApiController
+    def available_packages
```

Comment:
any objections to just putting all of this in the existing packages index action?  It already supports applicable, upgradable, host_id, and repository_id.  You'd just need to add some param 'available_for', as we've done in other places:  https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/host_collections_controller.rb#L34-L35
or
https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/products_controller.rb#L40

and then just put the logic in the 'indexed_relation' method.

Reply:
oh i see, this is already a thing for the errata controller,  thoughts?